ethcore/src/engines/clique/mod.rs

```diff
@@ -405,8 +404,12 @@ impl Engine<EthereumMachine> for Clique {
 		let mut seal: Vec<u8> = Vec::with_capacity(VANITY_LENGTH + SIGNATURE_LENGTH);
 
 		// At this point, extra_data should only contain miner vanity.
-		if header.extra_data().len() > VANITY_LENGTH {
-			panic!("on_seal_block: unexpected extra_data: {:?}", header);
+		if header.extra_data().len() != VANITY_LENGTH {
```

Comment:
This is more accurate because `VANITY_DATA` should be 32 bytes but it may be out-of-scope for this PR